src/components/version-redirect.jsx

```diff
@@ -0,0 +1,36 @@
+import React from 'react';
+import {Redirect} from 'react-router-dom';
+import Loading from './loading';
+
+export default class VersionRedirect extends React.Component {
+  constructor (props) {
+    super (props);
+    this.state = {
+      pageId: null,
+      error: null
+    };
+  }
+
+  componentDidMount () {
+    const versionId = this.props.match.params.versionId;
+    this.props.api.getVersionById(versionId)
```

Comment:
I completely missed this before, sorry. `api` should be coming from `context`, where it’s already available, not passed as an explicit prop. This should be the same as all the other components that use it.